snapcraft/_options.py

```diff
@@ -207,11 +207,14 @@ def get_core_dynamic_linker(self):
             if dynamic_linker_path in seen_paths:
                 raise SnapcraftEnvironmentError(
                     "found symlink loop resolving dynamic linker path")
+
             seen_paths.add(dynamic_linker_path)
-            if not os.path.exists(dynamic_linker_path):
+            if not (os.path.exists(dynamic_linker_path) or
```

Comment:
I don't think this is right, the correct fix is to replace os.path.exists with os.path.lexists. Sorry about this!